buzzfeedbotOS.py

```diff
@@ -19,45 +20,25 @@ def reddit_bot(headline, main_text, link):
 '''Checks if there are numbered bullet points in the article. If there is at least one, returns true, 
 if there are none, returns false''' 
 def check_for_numbered_points(link_to_check):
-	i=0
 	session = requests.Session()
 	clickbait_article = session.get(link_to_check)
 	soup = BeautifulSoup(clickbait_article.content, 'html.parser')
-	for title in soup.find_all('h3'):
-		for number in title.find_all('span', attrs={'class': 'subbuzz__number'}):
-			i+=1
-	if i > 0:
-		return (True)
-	else:
-		return (False)
+	return any(title.find_all('span', attrs={'class': 'subbuzz__number'}) for title in soup.find_all('h3'))

```

Comment:
Note that using `any` here also ensures the loop is terminated early. Useful if there are many matches to both `find_all`s.

Reply:
Very true. I don't know why i bothered going through the whole loop if i was just looking for i>1.
Would it be OK to just put a break statement after i+=1 and then return i>0 as you said above?

Reply:
Nevermid what i said above. I think the best solution is to have return (True) within the second for loop, and return (False) outside and after both loops. Essentially getting rid of the i variable altogether